services/api/src/routes/products.js

```diff
@@ -56,7 +57,14 @@ router
   .patch('/:productId', validateBody(Product.getUpdateValidation()), async (ctx) => {
     const product = ctx.state.product;
     product.assign(ctx.request.body);
+
     await product.save();
+
+    await AuditLog.append('updated product', ctx, {
+      type: 'admin',
+      ...AuditLog.getObjectFields(product, ['shop', 'name']),
```

Comment:
This is sort of an example of what I was talking about above, we have a static method providing output that goes right back in as input... Unless `getObjectFields` is needed somewhere else it seems better to accept different input and remove some interface in the process.

Reply:
should be addressed now